packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js

```diff
@@ -0,0 +1,238 @@
+
+    const guestCheckout = () => {
+        if (isGuestCheckout) {
+            return (
+                <GuestCheckoutOptions
+                    classes={classes}
+                    handleSignIn={handleSignIn}
+                />
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const priceAdjustments = () => {
+        const showPriceAdjustments =
+            shippingInformationDone &&
+            shippingMethodDone &&
+            !paymentInformationDone;
+
+        if (showPriceAdjustments) {
+            return (
+                <div className={classes.price_adjustments_container}>
+                    <PriceAdjustments />
+                </div>
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const itemsReview = () => {
+        const showItemsReview =
+            shippingInformationDone &&
+            shippingMethodDone &&
+            paymentInformationDone;
+
+        if (showItemsReview) {
+            return (
+                <div className={classes.items_review_container}>
+                    <ItemsReview />
+                </div>
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const showOrderConfirmation = isCartEmpty && orderPlaced;
+
+    const orderConfirmation = useMemo(() => {
+        if (showOrderConfirmation) {
+            return <OrderConfirmationPage />;
+        } else {
+            return null;
+        }
+    }, [showOrderConfirmation]);
+
+    const emptyCart = () => {
+        if (isCartEmpty && !orderPlaced) {
+            return (
+                <EmptyCartMessage
+                    classes={classes}
+                    isGuestCheckout={isGuestCheckout}
+                />
+            );
+        } else {
+            return null;
+        }
+    };
+
+    const checkOutButton = () => {
+        const { place_order_button, review_order_button } = classes;
+        if (shippingInformationDone && shippingMethodDone) {
+            if (paymentInformationDone) {
+                return (
+                    <Button
+                        onClick={placeOrder}
+                        priority="high"
+                        className={place_order_button}
+                    >
+                        {'Place Order'}
+                    </Button>
+                );
+            } else {
+                return (
+                    <Button
+                        onClick={setPaymentInformationDone}
+                        priority="high"
+                        className={review_order_button}
+                    >
+                        {'Review Order'}
+                    </Button>
+                );
+            }
+        } else {
+            return null;
+        }
+    };
+
+    return (
+        <div className={classes.root}>
+            <Title>{`Checkout - ${STORE_NAME}`}</Title>
+            {!isCartEmpty ? (
```

Comment:
Here's another inline conditional to fixup.

Reply:
Left it on purpose. It is the whole component render output. Moving it into a variable and just returning that variable seamed weird. The return state will just have that variable.

Reply:
Done.